includes/class-payment-information.php

```diff
@@ -260,7 +274,16 @@ public static function from_payment_request(
 			$order->add_meta_data( 'is_woopay', true, true );
 			$order->save_meta_data();
 		}
-		return new Payment_Information( $payment_method, $order, $payment_type, $token, $payment_initiated_by, $manual_capture, $cvc_confirmation, $fingerprint, $payment_method_stripe_id );
+		$pi = new Payment_Information( $payment_method, $order, $payment_type, $token, $payment_initiated_by, $manual_capture, $cvc_confirmation, $fingerprint, $payment_method_stripe_id );
+
+		if ( self::PAYMENT_METHOD_ERROR === $payment_method ) {
+			$error_message = $request['wcpay-payment-method-error-message'] ?? __( 'Invalid Payment Method', 'woocommerce-payments' );
+			$error_code    = $request['wcpay-payment-method-error-code'] ?? 'unknown-error';
+			$error         = new \WP_Error( $error_code, $error_message );
+			$pi->set_error( $error );
+		}
+
+		return $pi;
```

Comment:
```suggestion
		$payment_information = new Payment_Information( $payment_method, $order, $payment_type, $token, $payment_initiated_by, $manual_capture, $cvc_confirmation, $fingerprint, $payment_method_stripe_id );

		if ( self::PAYMENT_METHOD_ERROR === $payment_method ) {
			$error_message = $request['wcpay-payment-method-error-message'] ?? __( 'Invalid Payment Method', 'woocommerce-payments' );
			$error_code    = $request['wcpay-payment-method-error-code'] ?? 'unknown-error';
			$error         = new \WP_Error( $error_code, $error_message );
			$payment_information->set_error( $error );
		}

		return $payment_information;
```

⛏️ `$pi` might be a bit on the short side. While it works, it might be clearer for others (and our future selves) if we used more descriptive names, like `$payment_information` instead. I think it could make the code easier to understand and maintain, even when it's in a short scope like this. What do you think?